catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/QueryResultIterable.java

```diff
@@ -131,9 +136,17 @@ private void queryNext(int index) {
                 Map<String, Serializable> props = new HashMap<>();
                 // Avoid caching all results while dumping with native query mode
                 props.put("mode", "native");
-                response = catalog.query(filter.apply(index));
+                response = catalog.query(new QueryRequestImpl(new QueryImpl(filter,
+                        pageIndex,
+                        pageSize,
+                        SortBy.NATURAL_ORDER,
+                        false,
+                        5000l/*timeout? config? passed in?*/)));
+                missingResults = pageSize - response.getResults()
+                        .size();
+
             } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
-                throw new CatalogCommandRuntimeException(e);
+                throw new CatalogRuntimeException(e);
             }
             List<Result> queryResults = response.getResults();
             this.results = queryResults.iterator();
```

Comment:
❗️ We can't just check if `size < pageSize` since it's possible that `size` is smaller because that's all the catalog framework will return, and not because there's nothing else.